hydrolib/hydromt_delft3dfm/workflows/branches.py

```diff
@@ -223,6 +225,14 @@ def cleanup_branches(
     n = np.sum(list(branches_length <= 0.1))
     branches = branches[branches_length >= 0.1]
     logger.debug(f"Removing {n} branches that are shorter than 0.1 meter.")
+    # remove branches with ring geometries
+    first_nodes = [l.coords[0] for l in branches.geometry]
+    last_nodes = [l.coords[-1] for l in branches.geometry]
+    duplicate_ids = np.isclose(first_nodes, last_nodes)
+    duplicate_ids = [
+        branches.index[i] for i in range(len(branches)) if np.all(duplicate_ids[i])
+    ]
+    branches = branches.drop(duplicate_ids, axis=0)
```

Comment:
Since the function is already pretty long you could extract this to a new method -> 

```
def _remove_branches_with_ring_geometries(branches: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
    first_nodes = [l.coords[0] for l in branches.geometry]
    last_nodes = [l.coords[-1] for l in branches.geometry]
    duplicate_ids = np.isclose(first_nodes, last_nodes)
    duplicate_ids = [
        branches.index[i] for i in range(len(branches)) if np.all(duplicate_ids[i])
    ]
    branches = branches.drop(duplicate_ids, axis=0)
    logger.debug("Removing branches with ring geometries.")
    
    return branches
```

Reply:
Check